docs/source/input_output.rst

```diff
@@ -73,6 +73,8 @@ The latest complete checkpoint version is stored in an additional file in the sa
 
 Each checkpoint contains a JSON dump of the config that was used to produce it stored in the :file:`config.json` file.
 
+Old checkpoints will be automatically deleted unless specified through ``checkpoint_interval``
```

Comment:
Could you make this clearer? Ideally, the doc should explain more than the `--help` text.

Like: "When a new checkpoint is saved, the previous one is deleted. If ``checkpoint_interval`` is specified, one checkpoint every ``checkpoint_interval`` epochs is preserved."

PS: `checkpoint_preservation_interval` is a longer but more self-descriptive name for that field.